### PR TITLE
Improve example readability

### PR DESCRIPTION
**Pull request by keli-wen**
## Description

Improve example readability.  Added comments and adjusted the code structure appropriately.

The executing log:
```
$ python examples/multigpu/node_classification_sage.py
Loading data
Training in mixed mode using 1 GPU(s)
Epoch 00000 | Loss 1.3259 | Accuracy 0.8552 | Time 21.5105
Epoch 00001 | Loss 0.5896 | Accuracy 0.8770 | Time 20.7160
Epoch 00002 | Loss 0.4923 | Accuracy 0.8861 | Time 20.7272
Epoch 00003 | Loss 0.4472 | Accuracy 0.8904 | Time 20.7882
Epoch 00004 | Loss 0.4185 | Accuracy 0.8955 | Time 20.7401
Epoch 00005 | Loss 0.3973 | Accuracy 0.8960 | Time 20.7272
Epoch 00006 | Loss 0.3852 | Accuracy 0.8977 | Time 20.7231
Epoch 00007 | Loss 0.3765 | Accuracy 0.9000 | Time 20.7235
Epoch 00008 | Loss 0.3630 | Accuracy 0.9011 | Time 20.7510
Epoch 00009 | Loss 0.3569 | Accuracy 0.9035 | Time 20.7545
Epoch 00010 | Loss 0.3494 | Accuracy 0.9025 | Time 20.6905
Epoch 00011 | Loss 0.3477 | Accuracy 0.9042 | Time 20.7988
Epoch 00012 | Loss 0.3393 | Accuracy 0.9044 | Time 20.6936
Epoch 00013 | Loss 0.3387 | Accuracy 0.8992 | Time 20.7619
Epoch 00014 | Loss 0.3386 | Accuracy 0.9063 | Time 20.7725
Epoch 00015 | Loss 0.3285 | Accuracy 0.9062 | Time 20.7332
Epoch 00016 | Loss 0.3258 | Accuracy 0.9065 | Time 20.7991
Epoch 00017 | Loss 0.3270 | Accuracy 0.8982 | Time 20.8652
Epoch 00018 | Loss 0.3309 | Accuracy 0.9063 | Time 20.7522
Epoch 00019 | Loss 0.3219 | Accuracy 0.9063 | Time 20.7846
100%|█████████████████████████████| 2392/2392 [00:17<00:00, 134.98it/s]
100%|██████████████████████████████| 2392/2392 [00:29<00:00, 80.88it/s]
100%|██████████████████████████████| 2392/2392 [00:29<00:00, 81.04it/s]
Test accuracy 0.7738
```

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: examples/multigpu/node_classification_sage.py [hunk 1]
```diff
@@ -23,22 +23,66 @@
...
+        self.dropout = nn.Dropout(0.5)
+
     def forward(self, blocks, x):
+        """
+        Forward propagation of the SAGE model.
+
+        Parameters:
+        -----------
+        blocks : list of dgl.Block objects
+            List of blocks that corresponds to the computation dependency of the GraphSAGE layers.
```
Comment (frozenbugs) [inline examples/multigpu/node_classification_sage.py:67]: max char 80.

File: examples/multigpu/node_classification_sage.py [hunk 2]
```diff
@@ -89,14 +133,45 @@ def inference(self, g, device, batch_size, use_uva):
...
     y_hats = []
+
     for it, (input_nodes, output_nodes, blocks) in enumerate(dataloader):
         with torch.no_grad():
             x = blocks[0].srcdata["feat"]
             ys.append(blocks[-1].dstdata["label"])
             y_hats.append(model(blocks, x))
+
+    # concatenate all true labels and all predictions.
+    # then calculate the accuracy of the predictions using the function MF.accuracy
```
Comment (frozenbugs) [inline examples/multigpu/node_classification_sage.py:174]:
```suggestion
    # Concatenate all true labels and all predictions, then calculate the
    # accuracy of the predictions using the function MF.accuracy
```

File: examples/multigpu/node_classification_sage.py (hunk 1 above)
Comment (frozenbugs) [reply examples/multigpu/node_classification_sage.py:67]: and others.

File: examples/multigpu/node_classification_sage.py [hunk 3]
```diff
@@ -222,11 +388,56 @@ def run(proc_id, nprocs, devices, g, data, mode, num_epochs):
...
     )
+    # after training, perform inference (i.e., make predictions) on the test data
     layerwise_infer(proc_id, device, g, num_classes, test_idx, model, use_uva)
+
     # cleanup process group
     dist.destroy_process_group()
 
 
+def main(args):
+    # get the device list
```
Comment (frozenbugs) [inline examples/multigpu/node_classification_sage.py:399]: For all comment sentences: capitalize the first char, and end with dot.

File: examples/multigpu/node_classification_sage.py [hunk 4]
```diff
@@ -132,9 +208,51 @@ def train(
...
+        If True, uses Unified Virtual Addressing (UVA) for CUDA computation.
+
+    num_epochs : int
+        The number of epochs for which to train the model.
+
+    Returns:
+    --------
+    None
+    """
+    # Define the neighborhood sampler. The arguments [10, 10, 10] specify that we sample
```
Comment (frozenbugs) [inline examples/multigpu/node_classification_sage.py:250]: 80 chars max

File: examples/multigpu/node_classification_sage.py [hunk 5]
```diff
@@ -132,9 +209,52 @@ def train(
...
+    None
+    """
+    # Define the neighborhood sampler. The arguments [10, 10, 10]
+    # specify that we sample 10 neighbors at each layer for 3 layers.
+    # Prefetching node features and labels for speedup.
     sampler = NeighborSampler(
         [10, 10, 10], prefetch_node_feats=["feat"], prefetch_labels=["label"]
     )
+
     train_dataloader = DataLoader(
```
Comment (frozenbugs) [inline examples/multigpu/node_classification_sage.py:258]: This is a place that might cause confusion for users. The dataloader was created in different processs but the initiation does not take proc_id as parameter. It worth to callout how it was handled.

File: examples/multigpu/node_classification_sage.py [hunk 6]
```diff
@@ -177,6 +299,8 @@ def train(
             evaluate(model, g, num_classes, val_dataloader).to(device) / nprocs
         )
         t1 = time.time()
+
+        # Reduce validation accuracy from all processes.
         dist.reduce(acc, 0)
         if proc_id == 0:
             print(
```
Comment (frozenbugs) [inline examples/multigpu/node_classification_sage.py:306]: use fstring.

File: examples/multigpu/node_classification_sage.py [hunk 7]
```diff
@@ -222,11 +390,56 @@ def run(proc_id, nprocs, devices, g, data, mode, num_epochs):
         use_uva,
         num_epochs,
     )
+    # After training, perform inference on the test data.
     layerwise_infer(proc_id, device, g, num_classes, test_idx, model, use_uva)
```
Comment (frozenbugs) [inline examples/multigpu/node_classification_sage.py:394]: Why this is named layerwise_infer? What does layerwise refer to?

File: examples/multigpu/node_classification_sage.py [hunk 8]
```diff
@@ -222,11 +390,56 @@ def run(proc_id, nprocs, devices, g, data, mode, num_epochs):
...
-    # cleanup process group
+
+    # Cleanup process group.
     dist.destroy_process_group()
 
 
+def main(args):
+    # Get the device list
```
Comment (frozenbugs) [inline examples/multigpu/node_classification_sage.py:401]:
```suggestion
    # Get the device list.
```

File: examples/multigpu/node_classification_sage.py (hunk 7 above)
Comment (keli-wen) [reply examples/multigpu/node_classification_sage.py:394]: The function name `layerwise_infer` is probably due to the fact that in the model inference phase, the inference process is performed layer-wise. In graph neural networks (e.g. GraphSAGE), information is propagated through the layers of the network. 

But the naming of `layerwise_infer` is prevalent in all files related to `GraphSAGE` model, e.g. `graphsage.py`, so if I need to add an explanation of the function naming, I might need to go and add comments to all sections where `layerwise_infer` appears?

**Pull request closed without merging**